File: Euclid/Messages/Outgoing/FLATINFO.cs
using Euclid.Game;
using Euclid.Network.Streams.Util;
using Euclid.Storage.Database.Data;

namespace Euclid.Messages.Outgoing
{
    class FLATINFO : IMessageComposer
    {
        private Room room;

        public FLATINFO(Room room)
        {
            this.room = room;
        }

        public override void Write()
        {
            this.Data.Add(new KeyValueEntry("name", room.Data.Name, "="));
            this.Data.Add(new KeyValueEntry("password", room.Data.Password, "="));
            this.Data.Add(new KeyValueEntry("description", room.Data.Description, "="));
            this.Data.Add(new KeyValueEntry("showOwnerName", room.Data.ShowName ? "true" : "false", "="));
            this.Data.Add(new KeyValueEntry("allsuperuser", room.Data.SuperUsers ? "true" : "false", "="));

            if (room.Data.AccessType == RoomStatus.OPEN)
                Data.Add(new KeyValueEntry("doormode", "open", "="));

            if (room.Data.AccessType == RoomStatus.CLOSED)
                Data.Add(new KeyValueEntry("doormode", "closed", "="));

            if (room.Data.AccessType == RoomStatus.PASSWORD)
                Data.Add(new KeyValueEntry("doormode", "password", "="));
        }
    }
}

File: euclid/src/main/java/euclid/messages/outgoing/FLATINFO.java
package euclid.messages.outgoing;

import euclid.game.room.Room;
import euclid.messages.IMessageComposer;
import euclid.network.streams.util.KeyValueEntry;
import euclid.storage.database.data.RoomStatus;

public class FLATINFO extends IMessageComposer {

    private final Room room;

    public FLATINFO(Room room) {
        this.room = room;
    }

    @Override
    public void write() {
        getData().add(new KeyValueEntry("name", room.getData().getName(), "="));
        getData().add(new KeyValueEntry("password", room.getData().getPassword(), "="));
        getData().add(new KeyValueEntry("description", room.getData().getDescription(), "="));
        getData().add(new KeyValueEntry("showOwnerName", room.getData().isShowName() ? "true" : "false", "="));
        getData().add(new KeyValueEntry("allsuperuser", room.getData().isSuperUsers() ? "true" : "false", "="));

        if (room.getData().getAccessType() == RoomStatus.OPEN) {
            getData().add(new KeyValueEntry("doormode", "open", "="));
        }
        if (room.getData().getAccessType() == RoomStatus.CLOSED) {
            getData().add(new KeyValueEntry("doormode", "closed", "="));
        }
        if (room.getData().getAccessType() == RoomStatus.PASSWORD) {
            getData().add(new KeyValueEntry("doormode", "password", "="));
        }
    }
}
